# On serpent, the ncurses header lives in /usr/include/ncurses, so fix that.
#
# In addition, don't attempt to link against libtinfo (-ltinfo) as ncurses.so
# already contains the necessary symbols.
#
diff --git a/Include.make b/Include.make
index ff08451..7b706f9 100644
--- a/Include.make
+++ b/Include.make
@@ -4,5 +4,5 @@ USRSBINDIR      := /usr/sbin
 USRLIBDIR       := /usr/lib
 MAN8            := /usr/share/man/man8
 ifndef CFLAGS
-CFLAGS          := -Wall -O2
+CFLAGS          := -Wall -O2 -I/usr/include/ncurses/
 endif
diff --git a/usr_sbin/Makefile b/usr_sbin/Makefile
index 6beb2b0..e09d40c 100644
--- a/usr_sbin/Makefile
+++ b/usr_sbin/Makefile
@@ -9,8 +9,8 @@ all: libtomoyotools.so $(BUILD_FILES)
 
 $(BUILD_FILES): libtomoyotools.so
 
-/usr/include/ncurses.h:
-	@echo "/usr/include/ncurses.h is missing."
+/usr/include/ncurses/ncurses.h:
+	@echo "/usr/include/ncurses/ncurses.h is missing."
 	@echo "Run 'yum install ncurses-devel' or 'apt-get install libncurses-dev'"
 	sleep 10
 
@@ -25,12 +25,12 @@ libtomoyotools.so: tomoyotools.c tomoyotools.h
 .c:
 	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $< -ltomoyotools -L.
 
-tomoyo-editpolicy: tomoyotools.h editpolicy*.c readline.h /usr/include/ncurses.h libtomoyotools.so
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o tomoyo-editpolicy editpolicy*.c -lncurses -ltinfo -ltomoyotools -L. -DCOLOR_ON -DNCURSES_WIDECHAR=0 || \
+tomoyo-editpolicy: tomoyotools.h editpolicy*.c readline.h /usr/include/ncurses/ncurses.h libtomoyotools.so
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o tomoyo-editpolicy editpolicy*.c -lncurses -ltomoyotools -L. -DCOLOR_ON -DNCURSES_WIDECHAR=0 || \
 	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o tomoyo-editpolicy editpolicy*.c -lncurses -ltomoyotools -L. -DCOLOR_ON -DNCURSES_WIDECHAR=0
 
-tomoyo-queryd: tomoyotools.h tomoyo-queryd.c readline.h /usr/include/ncurses.h libtomoyotools.so
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o tomoyo-queryd tomoyo-queryd.c -lncurses -ltinfo -ltomoyotools -L. -DNCURSES_WIDECHAR=0 || \
+tomoyo-queryd: tomoyotools.h tomoyo-queryd.c readline.h /usr/include/ncurses/ncurses.h libtomoyotools.so
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o tomoyo-queryd tomoyo-queryd.c -lncurses -ltomoyotools -L. -DNCURSES_WIDECHAR=0 || \
 	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o tomoyo-queryd tomoyo-queryd.c -lncurses -ltomoyotools -L. -DNCURSES_WIDECHAR=0
 
 install: all
